### ISSUE TITLE
local scope problems with @parallel

### ISSUE THREAD

**Issue by EthanAnderes** [JuliaLang/julia#6760]
The call to `@parallel` in the local scope below can't find `a`

```
addprocs(3)
if true
    a = 2
    x = @parallel (vcat) for k=1:2
        sin(a)
    end  #<--- gives an error
end
```

Outside the conditional scope it works fine

```
addprocs(3)
a = 2
x = @parallel (vcat) for k=1:2
    sin(a)
end   #<---works fine
```

[google groups reference](https://groups.google.com/forum/m/?fromgroups#!topic/julia-users/-Y1rc8gkrgo)

[pao: quoting macro name]

**Comment by amitmurthy** [comment]
Closed by #19594

**Comment by tkelman** [comment]
is this case tested?
